docs/releases/5.2.md

```diff
@@ -107,6 +107,49 @@ depth: 1
 
 Wagtail now uses the Willow feature of providing the image MIME type, which uses the [`filetype`](https://pypi.org/project/filetype/) package.
 
+### Adoption of `classname` convention for `MenuItem` related classes and hooks
+
+Wagtail `MenuItem` and menu hooks have been updated to use the more consistent naming of `classname` (singular) instead of `classnames` (plural), a convention that started in Wagtail 4.2.
+
+The current `classnames` keyword argument naming will be supported, but will trigger a deprecation warning. Support for this variant will be removed in a future release.
+
+The following classes will adopt this new convention.
+
+-   `admin.menu.MenuItem`
+-   `admin.ui.sidebar.ActionMenuItem`
+-   `admin.ui.sidebar.LinkMenuItem`
+-   `admin.ui.sidebar.PageExplorerMenuItem`
+-   `contrib.settings.registry.SettingMenuItem`
+
+The following hooks usage may be impacted if `classnames` were used when generating menu items.
+
+-   `register_admin_menu_item`
+-   `register_settings_menu_item`
+
+### Example
+
+```python
+from django.urls import reverse
+
+from wagtail import hooks
+from wagtail.admin.menu import MenuItem
+
+@hooks.register('register_admin_menu_item')
+def register_frank_menu_item():
+  return MenuItem(
+    'Frank',
+    reverse('frank'),
+    icon_name='folder-inverse',
+    order=10000,
+    classname="highlight-menu" # not classnames=...
+  )
+```
+
+| Usage                                   | Old (`classnames`) | New (`classname`) |
+| --------------------------------------- | ------------------ | ----------------- |
+| `register_admin_menu_item` (hook usage) | `...`              | `...`             |
+| `register_settings_menu_item` (hook usage) | `...`              | `...`             |
```

Comment:
I will get rid of this table, I originally was going to do multiple examples but I think the one code example above is enough.